src/App.tsx

```diff
@@ -1,26 +1,210 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  USER_ID,
+  addTodo,
+  getTodos,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import Header from './Cpmponents/Header';
+import TodoList from './Cpmponents/TodoList';
+import Footer from './Cpmponents/Footer';
+import { Status } from './types/status';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodos, setTempTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(Status.all);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [newTodo, setNewTodo] = useState('');
+  const [inProcess, setInProcess] = useState<number[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    setErrorMessage('');
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setTimeout(() => setErrorMessage(''), 3000);
+      });
+  }, []);
+
+  const Tasks = () => {
+    const { active, completed } = Status;
+    const allTodos = [...todos, ...tempTodos];
+
+    switch (filterBy) {
+      case active:
+        return allTodos.filter(todo => !todo.completed);
+
+      case completed:
+        return allTodos.filter(todo => todo.completed);
+
+      default:
+        return allTodos;
+    }
+  };
+
+  const allActive = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+
+  const completedTask = todos.length > 0 && allActive === 0;
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const changeFilter = useCallback((newFilter: Status) => {
```

Comment:
this function wrapper seems redundant - you should just use `setFilterBy` function instead